cxx/hirm.hh

```diff
@@ -753,13 +785,29 @@ class IRM {
       domain_to_relations.at(d).insert(name);
       doms.push_back(domains.at(d));
     }
-    relations[name] = new Relation(name, relation.distribution, doms, prng);
+    if (relation.distribution == "normal") {
```

Comment:
Lines 788-798 should be their own standalone function, get_relation_by_name or something like that.

Reply:
I'll factor this out after Srin moves definitions to `.cc` files, I was getting "multiple definition" errors.